### PR TITLE
Add test cases for expose API to the App developers(1.5)

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
## Purpose
Visibility of APIs to the App developers restricted by roles and multiple roles
Visibility of APIs to the public App developers without any restriction

### CODE REVIEW

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/resources/testng.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<suite name="ApiManager-scenario-tests-suite">
+    <test name="1.1.5-create-edit-rest-api" preserve-order="true" verbose="2" parallel="false">
+        <classes>
+            <class name="org.wso2.am.scenario.tests.rest.api.publicVisibility.PublicRestApiVisibilityTestCase"/>
+            <!--<class name="PublicRestApiVisibilityNegativeTestCase"/>-->
```
Comment (mushthaq33) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/resources/testng.xml]: Lets remove the commented out line.

File: product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java [hunk 2]
```diff
@@ -85,6 +86,30 @@ public APIRequest(String name, String context, String visibility, String version
         constructSwagger();
     }
 
+    public APIRequest(String name, String context, String visibility, String roles, String version, String resource, String tiersCollection, URL endpointUrl) {
```
Comment (mushthaq33) [inline product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java]: This line seems to exceed 120 characters. Please check.

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+    private void validateRoles(String roles) throws APIManagerIntegrationTestException {
+
+        HttpResponse checkValidationRole = apiPublisher.validateRoles(roles);
+        assertTrue(checkValidationRole.getData().contains("true"));
+        verifyResponse(checkValidationRole);
+    }
+
+    private void createAPI(APIRequest apiCreationRequest, String apiName) throws APIManagerIntegrationTestException {
+
```
Comment (mushthaq33) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: Lets remove these type of new lines.

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    private void createRole(UserManagementClient userManagementClient, String role) throws Exception {
+
+        userManagementClient.addRole(role,
+                new String[]{},
+                new String[]{"/permission/admin/login",
+                        "/permission/admin/manage/api/subscribe"});
+    }
+
+    private void createUser(UserManagementClient userManagementClient, String userName, String password, String role) throws Exception {
```
Comment (mushthaq33) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: Exceeding 120 characters

File: product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java [hunk 5]
```diff
@@ -85,6 +86,30 @@ public APIRequest(String name, String context, String visibility, String version
...
+
+        this.name = name;
+        this.context = context;
+        this.version = version;
+        this.resource = resource;
+        this.visibility = visibility;
+        this.tiersCollection = tiersCollection;
+
+        if (this.visibility == "restricted"){
```
Comment (mushthaq33) [inline product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java]: Formatting issue

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/logs/automation.log [hunk 6]
```diff
@@ -0,0 +1,1114 @@
+2018-12-11 17:25:57,044 WARN  [org.wso2.am.scenario.test.common.ScenarioTestBase] - Deployment property file not found: null/deployment.properties
```
Comment (Krishanx92) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/logs/automation.log]: We don't need to commit the logs

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+    @AfterTest(alwaysRun = true)
+    public void destroy() throws Exception {
+
+        apiPublisher.deleteAPI("PhoneVerificationAdd", apiVersion, ADMIN_LOGIN_USERNAME);
+        apiPublisher.deleteAPI("APIWildCardApi", apiVersion, ADMIN_LOGIN_USERNAME);
+
+        userManagementClient1.deleteUser("SubscriberUser");
+        userManagementClient1.deleteRole("Health-Subscriber");
+
+        userManagementClient2.deleteUser("MultipleRoleUser");
```
Comment (fazlan-nazeem) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: Why do we need 2 userManagement Clients? We can use just 1 right?

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.1-expose-api-visible-to-all-public-app-developer-without-restriction/pom.xml [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.am</groupId>
+            <artifactId>scenarios-common</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+    </dependencies>
+
+</project>
```
Comment (TanyaM) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.1-expose-api-visible-to-all-public-app-developer-without-restriction/pom.xml]: New line required

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/resources/testng.xml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<suite name="ApiManager-scenario-tests-suite">
+    <test name="1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles" preserve-order="true" verbose="2" parallel="false">
+        <classes>
+            <class name="org.wso2.am.scenario.tests.rest.api.restrictedVisibility.RESTApiVisibilityRestrictedByRolesNegativeTestCase"/>
+            <class name="org.wso2.am.scenario.tests.rest.api.restrictedVisibility.RestApiVisibilityRestrictedByRolesTestCase"/>
+        </classes>
+    </test>
+</suite>
```
Comment (TanyaM) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/resources/testng.xml]: New line required

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.1-expose-api-visible-to-all-public-app-developer-without-restriction/pom.xml (hunk 8 above)
Comment (HiranyaKavishani) [reply product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.1-expose-api-visible-to-all-public-app-developer-without-restriction/pom.xml]: Fixed with 1ced581ac554b27ba080eadb673460960d8a9381

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/resources/testng.xml (hunk 9 above)
Comment (HiranyaKavishani) [reply product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/resources/testng.xml]: Fixed with 1ced581ac554b27ba080eadb673460960d8a9381

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.1-expose-api-visible-to-all-public-app-developer-without-restriction/pom.xml (hunk 8 above)
Comment (HiranyaKavishani) [reply product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.1-expose-api-visible-to-all-public-app-developer-without-restriction/pom.xml]: Fixed with 1ced581ac554b27ba080eadb673460960d8a9381

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/resources/testng.xml (hunk 9 above)
Comment (HiranyaKavishani) [reply product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/resources/testng.xml]: Fixed with 1ced581ac554b27ba080eadb673460960d8a9381

**Pull request merged**